lib/ruby_lsp/listeners/document_highlight.rb

```diff
@@ -99,6 +115,12 @@ def initialize(response_builder, target, parent, dispatcher)
 
         return unless target && parent
 
+        if DEFINITION_AND_CONTROL_NODES.include?(target.class)
+          add_highlight_for_definition_and_control_node(target)
+
+          return
+        end
```

Comment:
This approach doesn't quite fit the listener architecture and it's a bit odd to be returning early in the initialize with a response.

We will need to ensure that we have listener events for all of the desired nodes and then in each one of them we add the highlight if applicable.

Reply:
No, we wouldn't want to highlight all of them. Only the one under the cursor. It would need to be something like this:

```ruby
def on_class_node_enter(node)
  if @target == node && position_covers_def_keyword
    # highlight `def` and `end`
  end
end
```

Currently, the document highlight listener doesn't have access to the position, but we can pass that in the initializer.

Reply:
ok, I’ll proceed with this approach.